src/dataframe/dataframe.jl

```diff
@@ -749,6 +765,10 @@ merge!(df, df2)  # column z is added, column id is overwritten
 """
 function Base.merge!(df::DataFrame, others::AbstractDataFrame...)
     for other in others
+        notinother = setdiff(names(other), names(df))
```

Comment:
It would probably be cleaner to define `merge!(::MetaData, ::MetaData...)`.

Reply:
I was trying to make all the code in `metadata.jl` not know anything about the dataframe attached to it. Since this code is really about working with the distinct names of both dataframes, i put it here.

Reply:
OK. I'm a bit concerned about the fact that this allocates, even when one isn't using meta-data at all. We should really ensure there's a minimal or zero overhead in that case. Maybe we can handle that by checking whether `metadata` is empty first?

In terms of code organization, maybe this could be moved to a function taking `Index` objects for two data frames, and it could also be used for `join` operations?